apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/ShardDownloader.java

```diff
@@ -156,10 +156,10 @@ public FileDownloadDecision prepareAndStartDownload() {
 
         } else {
             //we have some shards available on the networks, let's decide what to do
-            List<Long> shardIds = new ArrayList(sortedShards.entrySet());
+            List<Long> shardIds = new ArrayList(sortedShards.keySet());
             Collections.sort(shardIds);
             Long lastShard = shardIds.get(shardIds.size() - 1);
-            log.debug("Last known ShardId '{}'", lastShard);
+            log.debug("Last known ShardId '{}'", lastShard);            
```

Comment:
**lastShard** is = NULL when there is not any shard yet in network.

Reply:
`shardIds.size() - 1` can fail IndexOutOfBound exception if shardIds.size() = 0; // NO any shards yet in networks

Reply:
It is checked abouve and not empty